### DISCUSSION TITLE
请教一下，有什么合适的方法将一个pagfile中的帧替换给另一个pagfile？

### DISCUSSION THREAD

**Discussion by zgglin** [Tencent/libpag#1763] [Q&A] [answered]
hello 
当前我想把一个pagfile1（图片贴纸）的内容，作为另一个pagfile2（动画）的内容，来达到复用动画的目的。
加到同一个composition中达不到目的；
想通过pagfile2的replaceImage来操作，但是又找不到接口可以将pagfile1当前progress对应的帧转化为PAGImage
请教一下是否有其他合适的方法？ 谢谢解答！

**Comment by domchen** [comment, marked as answer]
你创建一个纹理，把它包装成PAGSurface，就可以给PAGPlayer绘制了，PAGPlayer.setProgress+flush逐帧刷新，内容就会到那个纹理上。然后同样这个纹理你包装成PAGImage再拿去替换就行。每帧都可以用相同的这个纹理。性能也很好。这是在Android上，要直接创建OpenGL纹理。IOS更简单一些，直接用CVPixelBuffer代替刚刚的纹理，还是一样的操作。
